frontend/www/js/omegaup/components/problem/Form.vue

```diff
@@ -275,19 +280,30 @@ export default class ProblemForm extends Vue {
     return JSON.stringify(this.selectedTags);
   }
 
-  get formattedVisibility(): number {
-    const visibility = Math.max(0, Math.min(2, this.visibility));
+  get formattedVisibility(): boolean {
     // when visibility is public warning, then the problem is shown as public
-    return visibility === 1 ? 2 : visibility;
+    return this.visibility > this.data.visibilityStatuses.private;
   }
 
-  set formattedVisibility(visibility: number) {
-    this.visibility = visibility;
+  set formattedVisibility(visibility: boolean) {
+    if (
+      this.originalVisibility === this.data.visibilityStatuses.publicWarning
+    ) {
+      this.visibility = visibility
+        ? this.data.visibilityStatuses.publicWarning
+        : this.data.visibilityStatuses.privateWarning;
+      return;
+    }
+    this.visibility = visibility
+      ? this.data.visibilityStatuses.public
+      : this.data.visibilityStatuses.private;
   }
 
-  get bannedOrPromoted(): boolean {
-    const visibility = Math.max(0, Math.min(2, this.visibility));
-    return visibility !== this.visibility;
+  get isEditable(): boolean {
+    return (
+      this.visibility < this.data.visibilityStatuses.promoted &&
+      this.visibility > this.data.visibilityStatuses.publicBanned
```

Comment:
```suggestion
      this.data.visibilityStatuses.publicBanned < this.visibility &&
      this.visibility < this.data.visibilityStatuses.promoted
```

es mejor usar el patrón `a < b && b < c` para que sea más obvio que se trata de que un número `b` está dentro del intervalo `(a, c)`.

Reply:
Listo